src/components/Pressable/Pressable.tsx

```diff
@@ -183,42 +243,90 @@ export default function Pressable(props: PressableProps) {
         })
         .onTouchesCancelled((event) => {
+          isPressCallbackEnabled.current = false;
+
           if (handlingOnTouchesDown.current) {
             cancelledMidPress.current = true;
-            onEndHandlingTouchesDown.current = () => pressOutHandler(event);
+            onEndHandlingTouchesDown.current = () =>
+              pressOutHandler(gestureTouchToPressableEvent(event));
             return;
           }
 
           if (
-            !isPressedDown.current ||
+            !hasPassedBoundsChecks.current ||
             event.allTouches.length > event.changedTouches.length
           ) {
             return;
           }
 
-          pressOutHandler(event);
+          pressOutHandler(gestureTouchToPressableEvent(event));
         }),
     [
+      activateLongPress,
+      longPressMinDuration,
       normalizedHitSlop,
-      props.unstable_pressDelay,
       pressInHandler,
       pressOutHandler,
+      props.unstable_pressDelay,
     ]
   );
 
-  // rippleGesture lives inside RNButton to enable android's ripple
-  const rippleGesture = useMemo(() => Gesture.Native(), []);
-
-  pressGesture.minDuration(
-    (props.delayLongPress ?? DEFAULT_LONG_PRESS_DURATION) +
-      (props.unstable_pressDelay ?? 0)
+  // ButtonGesture lives inside RNButton to enable android's ripple and to capture non-propagating events
+  const buttonGesture = useMemo(
+    () =>
+      Gesture.Native()
+        .onBegin(() => {
+          // Android sets BEGAN state on press down
+          if (Platform.OS === 'android') {
+            propagationGreenLight.current = true;
+          }
+        })
+        .onStart(() => {
+          // iOS sets ACTIVE state on press down
+          if (Platform.OS === 'ios') {
+            if (awaitingEventPayload.current) {
+              propagationGreenLight.current = true;
+              if (hasPassedBoundsChecks.current) {
+                pressInHandler(awaitingEventPayload.current);
+                awaitingEventPayload.current = null;
+              } else {
+                pressOutHandler(awaitingEventPayload.current);
+                propagationGreenLight.current = false;
+              }
+            } else {
+              if (hasPassedBoundsChecks.current) {
+                propagationGreenLight.current = true;
+              } else {
+                if (preventNativeEffects.current) {
+                  preventNativeEffects.current = false;
+                  return;
+                }
+
+                // In case Native Start is the very first gesture on iOS to launch
+                // that means we are in a no-scroll, no-nesting situation
+                // and native start may be trated as a replacement to the measure function.
+                propagationGreenLight.current = true;
+                isBoundsCheckNecessary.current = false;
+              }
+            }
+          }
+          if (Platform.OS === 'web') {
+            propagationGreenLight.current = true;
+          }
```

Comment:
We have to do something with this `if-else` hell 😅

Below you can find my suggestion, though I may have missed something.

```suggestion
          if (Platform.OS === 'web') {
            propagationGreenLight.current = true;
          }

          if (Platform.OS !== 'ios') {
            return;
          }

          if (awaitingEventPayload.current) {
            propagationGreenLight.current = true;

            if (hasPassedBoundsChecks.current) {
              pressInHandler(awaitingEventPayload.current);
              awaitingEventPayload.current = null;
            } else {
              pressOutHandler(awaitingEventPayload.current);
              propagationGreenLight.current = false;
            }

            return;
          }

          if (hasPassedBoundsChecks.current) {
            propagationGreenLight.current = true;
            return;
          }

          if (preventNativeEffects.current) {
            preventNativeEffects.current = false;
            return;
          }

          // In case Native Start is the very first gesture on iOS to launch
          // that means we are in a no-scroll, no-nesting situation
          // and native start may be trated as a replacement to the measure function.
          propagationGreenLight.current = true;
          isBoundsCheckNecessary.current = false;
```
```suggestion
          // iOS sets ACTIVE state on press down
          if (Platform.OS === 'ios') {
            if (awaitingEventPayload.current) {
              propagationGreenLight.current = true;
              if (hasPassedBoundsChecks.current) {
                pressInHandler(awaitingEventPayload.current);
                awaitingEventPayload.current = null;
              } else {
                pressOutHandler(awaitingEventPayload.current);
                propagationGreenLight.current = false;
              }
            } else {
              if (hasPassedBoundsChecks.current) {
                propagationGreenLight.current = true;
              } else {
                if (preventNativeEffects.current) {
                  preventNativeEffects.current = false;
                  return;
                }

                // In case Native Start is the very first gesture on iOS to launch
                // that means we are in a no-scroll, no-nesting situation
                // and native start may be trated as a replacement to the measure function.
                propagationGreenLight.current = true;
                isBoundsCheckNecessary.current = false;
              }
            }
          }
          if (Platform.OS === 'web') {
            propagationGreenLight.current = true;
          }
```

Reply:
I agree, I played around with flattening all of these `ifs` and ended up with something similar to the first suggestion, so I think the first suggestion makes the most sense.